### PR TITLE
docs: use `webpack serve` instead `webpack-dev-server`

### LINKED ISSUE #4095: Webpack Dev Server command deprecated
Hello,
The command for running webpack dev server on the [documentation page](https://webpack.js.org/guides/development/) is outdated.
Please see https://github.com/webpack/webpack-dev-server/issues/2029#issuecomment-585207461

It should be `webpack serve`, not `webpack-dev-server --open`.

Thank you
[ X] I am willing to work on this issue and submit a pull request.

EDIT: Sorry this command (`webpack server`)compiled my bundle, but deleted my `./dist` files. Will try to find the right command before the documentation is modified.

EDIT: `./dist` files deleted, but project is still served. `webpack serve` seems to still be correct.

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes https://github.com/webpack/webpack-cli/issues/1602#issuecomment-715273963
fixes #4095

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

File: src/content/configuration/dev-server.md [hunk 1]
```diff
@@ -210,20 +210,9 @@ module.exports = {
...
-
-`boolean`
-
-Enables/Disables colors on the console.
-
-```bash
-webpack-dev-server --color
-```
-
-
```
Comment (evilebottnawi) [inline src/content/configuration/dev-server.md:226]: Removed

File: src/content/configuration/dev-server.md [hunk 2]
```diff
@@ -1326,30 +1315,6 @@ module.exports = {
...
-};
-```
-
-Usage via the CLI
-
-```bash
-webpack-dev-server --socket socket
-```
-
-
```
Comment (evilebottnawi) [inline src/content/configuration/dev-server.md:1352]: Removed without replacement

File: src/content/configuration/dev-server.md [hunk 3]
```diff
@@ -527,7 +516,7 @@ module.exports = {
 Usage via the CLI
 
 ```bash
-webpack-dev-server --hot-only
+webpack serve --hot only
```
Comment (anshumanv) [inline src/content/configuration/dev-server.md:519]: `--hot-only` seems correct

Comment (evilebottnawi) [reply src/content/configuration/dev-server.md:519]: To be honestly we don't support this yet (regression), I am working on it, but --hot can be boolean or string, so we should supported:
- `webpack serve --hot`
- `webpack serve --no-hot`
- `webpack serve --hot=only` or `webpack serve --hot only`

Comment (anshumanv) [reply src/content/configuration/dev-server.md:519]: I see, though the option in devServer config object says `hotOnly` - https://webpack.js.org/configuration/dev-server/#devserverhotonly

This means we'll have to change/remove it's flag on `webpack-dev-server`? - https://github.com/webpack/webpack-dev-server/blob/98aa8e1f385b4e5d74ee097fa60362802cda505e/bin/cli-flags.js#L53

Comment (evilebottnawi) [reply src/content/configuration/dev-server.md:519]: Yes, we need to to do it

File: src/content/guides/development.md [hunk 4]
```diff
@@ -274,7 +274,7 @@ __package.json__
     "scripts": {
       "test": "echo \"Error: no test specified\" && exit 1",
       "watch": "webpack --watch",
-+     "start": "webpack-dev-server --open",
++     "start": "webpack serve --open",
```
Comment (chenxsan) [inline src/content/guides/development.md:277]: I got error `error: option '--open <value>' argument missing`, am I missing something?

Here's the versions:

```
    "webpack": "^5.2.0",
    "webpack-cli": "^4.1.0",
    "webpack-dev-server": "^3.11.0",
```

Comment (evilebottnawi) [reply src/content/guides/development.md:277]: Bug, I will fix it in the near future, but it will work

**Review by montogeek — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

File: src/content/configuration/dev-server.md [hunk 5]
```diff
@@ -136,7 +136,7 @@ module.exports = {
 To use this option with the CLI pass the `--allowed-hosts` option a comma-delimited string.
 
 ```bash
-webpack-dev-server --entry /entry/file --output-path /output/path --allowed-hosts .host.com,host2.com
+webpack serve --entry ./entry/file --output-path /output/path --allowed-hosts .host.com,host2.com
```
Comment (chenxsan) [inline src/content/configuration/dev-server.md:139]:
```suggestion
webpack serve --entry ./entry/file --output-path ./output/path --allowed-hosts .host.com,host2.com
```
Can we use relative path for `--output-path` just like `--entry`? I think it's more reasonable as we're dealing with the current working directory.

Comment (evilebottnawi) [reply src/content/configuration/dev-server.md:139]: Fixed

**Review by chenxsan — APPROVED** [review]

**Review by montogeek — APPROVED** [review]

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**